src/Illuminate/Database/Connectors/ConnectionFactory.php

```diff
@@ -96,8 +96,56 @@ protected function createReadPdo(array $config)
      */
     protected function createPdoResolver(array $config)
     {
+        if (array_key_exists('host', $config)) {
+            return $this->createPdoResolverWithHosts($config);
+        }
+        else{
+            return $this->createPdoResolverWithoutHosts($config);
+        }
+    }
+
+    /**
+     * Create a new Closure that resolves to a PDO instance with a specific host or an array of hosts.
+     *
+     * @param  array  $config
+     * @return \Closure
+     */
+    protected function createPdoResolverWithHosts(array $config){
+        return function () use ($config) {
+            if (!is_array($config['host'])) {
+                $hosts = [$config['host']];
+            } else {
+                $hosts = $config['host'];
+                shuffle($hosts);
+            }
+
+            foreach($hosts as $host){
+                $config['host'] = $host;
+
+                try{
+                    return $this->createConnector($config)->connect($config);
+                }
+                catch(\PDOException $e){
+                }
+            }
+
+            if (empty($hosts)) {
```

Comment:
If I move the check above the `foreach` loop it will be tested even if a connection was actually made, which seems redundant to me
It really is mostly a check so `$e` actually exists, since it's created inside the `foreach` loop